pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -154,12 +148,8 @@ def restore_model(self) -> None:
 
     def restore_model_weights(self, checkpoint_path: Optional[_PATH]) -> None:
         """Restore only the model weights."""
-        checkpoint = self._loaded_checkpoint
         if checkpoint_path is not None:
-            checkpoint = self._load_and_validate_checkpoint(checkpoint_path)
-
-        self.trainer.lightning_module.on_load_checkpoint(checkpoint)
-        self.trainer.training_type_plugin.load_model_state_dict(checkpoint)
```

Comment:
this means the state dict isn't loaded anymore. why this change?

Reply:
it gets called in new line 152 `_load_and_validate_checkpoint` --> `training_type_plugin.load_checkpoint` ?